### PR TITLE
Add pre-commit-config

### PR DESCRIPTION
**Pull request by redphix**
## What does this PR do?
Adds a [pre-commit](https://pre-commit.com/#install) config to run linters/formaters before committing 

Steps every contributor must follow before working:
1. make sure all the dependencies, include dev dependencies for python and node are installed
2. run `pip install pre-commit`
3. run `pre-commit install`
4. write awesome features, stage and commit your changes. The linters and formatters will run according to what you touch. If it's just a lint error (flake8) then fix it and stage the changes again and commit it.

## Checklist before merging
- [x] If it's a frontend feature, I have ran prettier `cd frontend; npm run format`. If it's a mobile app feature I ran `cd mobile; npm run format`.

### CODE REVIEW

File: api/logics.py [hunk 1]
```diff
@@ -734,7 +734,11 @@ def update_invoice(cls, order, user, invoice):
         if not order.taker_bond:
             return False, {"bad_request": "Wait for your order to be taken."}
         if (
-            not ( order.taker_bond.status == order.maker_bond.status == LNPayment.Status.LOCKED)
```
Comment (redphix) [inline api/logics.py:737]: Fixed the changes introduced by that 'annoy black' PR in this file

**Review by Reckless-Satoshi — COMMENTED** [review]
Very cool!

We probably should add those 4 steps at the top of `setup.md` as they apply to all contributors.

Maybe prettier can be fixed? I do not manage to trigger it. Tried changing the `files:` key too to no success.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  - repo: local
+    hooks:
+      - id: prettier-frontend
+        name: prettier-frontend
+        stages:
+          - commit
+          - merge-commit
+        language: system
+        files: ^frontend/
+        types: [javascript, jsx, ts, tsx, css, markdown, json] # uses https://github.com/pre-commit/identify
```
[the remark this answers is not in the document]
Comment (redphix) [reply .pre-commit-config.yaml]: ahh, was a mistake with the `types` key. Changed it to `types_or`

**Comment by Reckless-Satoshi** [comment]
Awesome, merging! :rocket:

We did not have a rewarded task created for this PR beforehand. But it's actually pretty important and will make or life easier. Let's add a few sats to the positive Karma, please submit an invoice for 100K Sats from a proxy wallet with a long expiration time.

**Comment by redphix** [comment]
@Reckless-Satoshi Thank you :heart: 

```
lnbc1m1p340jk0pp5pfj23dt0mwpscx2kzheqg6pass6sfa6jytegxtak8gptlzmg367sdqqcqzpgxqyd9uqsp52k2mkt4gqed6uwm529ytwvx30t2accpjf0et64gx708cd4f5p9ns9qyyssqpmg6wc63f4sjd0qw4sl5xtapg3qdj62t2tu9ja9j0y60talme5e9s0nzn2xct6e9n47u50wtmlsh4j5mvgdswv4pc5e9eqcxhdu7kwgqrlytu4

```

**Comment by Reckless-Satoshi** [comment]
> @Reckless-Satoshi Thank you heart hugs
> 
> ```
> lnbc1m1p340jk0pp5pfj23dt0mwpscx2kzheqg6pass6sfa6jytegxtak8gptlzmg367sdqqcqzpgxqyd9uqsp52k2mkt4gqed6uwm529ytwvx30t2accpjf0et64gx708cd4f5p9ns9qyyssqpmg6wc63f4sjd0qw4sl5xtapg3qdj62t2tu9ja9j0y60talme5e9s0nzn2xct6e9n47u50wtmlsh4j5mvgdswv4pc5e9eqcxhdu7kwgqrlytu4
> ```

fbe7afbb121b363e7f12fb2ed445116216698ef1a4dfb1be4e54c1f81cf9e5b4

**Pull request merged**